.github/workflows/links-fail-fast.yml

```diff
@@ -10,8 +10,16 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Restore lychee cache
+        uses: actions/cache@v3
+        if: always()
+        with:
+          path: .lycheecache
+          key: cache-lychee
+          restore-keys: cache-lychee-
+
       - name: Link Checker
         uses: lycheeverse/lychee-action@v1.9.3
         with:
           fail: true
-          args: --max-concurrency 5 .
+          args: --max-concurrency 5 --cache --max-cache-age 1d .
```

Comment:
After 1 day the cache would always return an out-dated `.lycheecache`.
See: https://github.com/open-telemetry/opentelemetry-go/actions/runs/8596315520/job/23552862599?pr=5160

I think we should "force-update" the cache. I see that there is a `save-always: true` option, but I am not sure if it does what we want. Maybe we would have to use `actions/cache/save@v3` ([docs](https://github.com/actions/cache/blob/main/save/README.md)).

Reply:
I think that in your run the key was different. I guess it was run then the key was generated from the SHA so it was always storing a new cache.

> Cache restored from key: cache-lychee-56719d07e5252211b258aa302f4a7e53da44f8de

Reply:
We can rerun the action from this PR in one day and see what happens 😉